cube/cube.cpp

```diff
@@ -703,69 +610,58 @@ void Demo::cleanup() {
     dfb->Release(dfb);
 #endif
 
-    inst.destroy(nullptr);
+    inst.destroy();
 }
 
 void Demo::create_device() {
-    float const priorities[1] = {0.0};
-
-    vk::DeviceQueueCreateInfo queues[2];
-    queues[0].setQueueFamilyIndex(graphics_queue_family_index);
-    queues[0].setQueueCount(1);
-    queues[0].setPQueuePriorities(priorities);
-
-    auto deviceInfo = vk::DeviceCreateInfo()
-                          .setQueueCreateInfoCount(1)
-                          .setPQueueCreateInfos(queues)
-                          .setEnabledLayerCount(0)
-                          .setPpEnabledLayerNames(nullptr)
-                          .setEnabledExtensionCount(enabled_extension_count)
-                          .setPpEnabledExtensionNames((const char *const *)extension_names)
-                          .setPEnabledFeatures(nullptr);
+    float priorities = {0.0};
+
+    std::vector<vk::DeviceQueueCreateInfo> queues;
+    queues.push_back(vk::DeviceQueueCreateInfo().setQueueFamilyIndex(graphics_queue_family_index).setQueuePriorities(priorities));
 
     if (separate_present_queue) {
-        queues[1].setQueueFamilyIndex(present_queue_family_index);
-        queues[1].setQueueCount(1);
-        queues[1].setPQueuePriorities(priorities);
-        deviceInfo.setQueueCreateInfoCount(2);
+        queues.push_back(
+            vk::DeviceQueueCreateInfo().setQueueFamilyIndex(present_queue_family_index).setQueuePriorities(priorities));
     }
 
-    auto result = gpu.createDevice(&deviceInfo, nullptr, &device);
-    VERIFY(result == vk::Result::eSuccess);
+    auto deviceInfo = vk::DeviceCreateInfo().setQueueCreateInfos(queues).setPEnabledExtensionNames(enabled_device_extensions);
+    auto device_return = gpu.createDevice(deviceInfo);
+    VERIFY(device_return.result == vk::Result::eSuccess);
+    device = device_return.value;
 }
 
-void Demo::destroy_texture(texture_object *tex_objs) {
+void Demo::destroy_texture(texture_object &tex_objs) {
     // clean up staging resources
-    device.freeMemory(tex_objs->mem, nullptr);
-    if (tex_objs->image) device.destroyImage(tex_objs->image, nullptr);
-    if (tex_objs->buffer) device.destroyBuffer(tex_objs->buffer, nullptr);
+    device.freeMemory(tex_objs.mem);
+    if (tex_objs.image) device.destroyImage(tex_objs.image);
+    if (tex_objs.buffer) device.destroyBuffer(tex_objs.buffer);
 }
 
 void Demo::draw() {
     // Ensure no more than FRAME_LAG renderings are outstanding
-    device.waitForFences(1, &fences[frame_index], VK_TRUE, UINT64_MAX);
+    device.waitForFences(fences[frame_index], VK_TRUE, UINT64_MAX);
     device.resetFences({fences[frame_index]});
 
-    vk::Result result;
+    auto acquire_result = vk::Result::eSuccess;
```

Comment:
why initialize?  The value here is immediately overwritten.

Reply:
fixed.